src/main/java/com/bravo/user/dao/specification/AddressSpecification.java

```diff
@@ -0,0 +1,19 @@
+package com.bravo.user.dao.specification;
+
+import com.bravo.user.dao.model.Address;
+import com.bravo.user.model.filter.AddressFilter;
+import lombok.RequiredArgsConstructor;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+@RequiredArgsConstructor
```

Comment:
Why are you using `@RequiredArgsConstructor` instead of `@AllArgsConstructor`?

Reply:
I just use it by default, not much of a difference here I don't think. In case some one comes along and forgets to make their fields `final` maybe this can sorta catch that.